agents/ppo.py

```diff
@@ -42,48 +48,57 @@ def create_from_config(cls, config):
             critic_loss_weight=config.get('critic_loss_weight', 1.0)
         )
 
-    def forward(self, state: State) -> Tuple[State, Policy, Value, Value]:
-        next_state, policy = self._actor.forward(state)
-        curr_v_func = self._critic.forward(state)  # (batch_size)
-        q_estimate = self._critic.forward(next_state)  # (batch_size)
-        # TODO [Vladimir Baikalov]: make same inerface for every agent's forward function
-        return next_state, policy, curr_v_func, q_estimate
+    def forward(self, inputs: Dict[str, Any]):
+        ppo_input = inputs[self._prefix]
+        next_state, policy = self._actor.forward(**ppo_input)
+        curr_v_func = self._critic.forward(**ppo_input)  # (batch_size)
+        q_estimate = self._critic.forward(**{**ppo_input, 'state': next_state})  # (batch_size)
```

Comment:
In this line, you do something strange. Since you update current_state to next_state it will not work because you don't update neighbors. Also, these is an inconsistency in terms of what you expect from critic to return (V function value or Q function value). Since you use the same critic it's kind of strange.

Reply:
Yes, old code, we decided simulation to update shared memory, so only next_state will be counted in forward)